File: Proyecto/Simulacion_PMSM/polos.m
% Definir el rango de tiempo
t = linspace(0, 10, 100); % de 0 a 10 segundos, con 100 puntos

% Inicializar matrices para almacenar los polos
all_poles = [];

for i = 1:length(t)
    % Definir los coeficientes del polinomio característico en función del tiempo t
    % Por ejemplo, un polinomio de tercer grado: a3*t^2 + a2*t + a1 + a0
    a3 = 1; % Coeficiente constante
    a2 = 2 * t(i); % Ejemplo: coeficiente lineal en t
    a1 = 3 * t(i)^2; % Ejemplo: coeficiente cuadrático en t
    a0 = 4; % Coeficiente constante

    % Crear el vector de coeficientes del polinomio
    p = [a3 a2 a1 a0];
    
    % Calcular las raíces del polinomio (los polos)
    poles = roots(p);
    
    % Almacenar los polos
    all_poles = [all_poles; poles.'];
end

% Graficar los polos en el plano complejo
figure;
hold on;
for i = 1:length(t)
    plot(real(all_poles(i,:)), imag(all_poles(i,:)), 'o');
end
xlabel('Parte Real');
ylabel('Parte Imaginaria');
title('Polos del Polinomio Característico en el Plano Complejo');
grid on;
hold off;

% Añadir una leyenda y etiquetas si es necesario
legend('Polos en diferentes tiempos');

File: Proyecto/Simulacion_PMSM/polos.py
import numpy as np
import matplotlib.pyplot as plt

def plot_poles(coefficients):
    """
    Grafica los polos de un polinomio característico dado por sus coeficientes.

    Parameters:
    coefficients (list or array-like): Coeficientes del polinomio característico,
                                       ordenados desde el término de mayor grado hasta el término independiente.
    """
    # Encontrar las raíces del polinomio (los polos)
    poles = np.roots(coefficients)
    
    # Configurar la figura y los ejes
    plt.figure(figsize=(8, 8))
    plt.axhline(0, color='black', linewidth=0.5)
    plt.axvline(0, color='black', linewidth=0.5)
    plt.grid(True, which='both', linestyle='--', linewidth=0.5)

    # Graficar los polos
    plt.plot(poles.real, poles.imag, 'x', markersize=10, label='Polos')

    # Añadir etiquetas y título
    plt.xlabel('Parte Real')
    plt.ylabel('Parte Imaginaria')
    plt.title('Polos del Polinomio Característico')
    plt.legend()
    plt.show()

# Ejemplo de uso
coefficients = [1, -3, 2]  # Coeficientes del polinomio característico: s^2 - 3s + 2
plot_poles(coefficients)
